pkg/baseimg/base_image.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2022, 2023 Chainguard, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package baseimg
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/chainguard-dev/go-apk/pkg/apk"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/layout"
+	ocitypes "github.com/google/go-containerregistry/pkg/v1/types"
+
+	"chainguard.dev/apko/pkg/build/types"
+)
+
+type BaseImage struct {
+	img      v1.Image
+	apkIndex []byte
```

Comment:
I'm not convinced we should persist here bytes or `[]*apk.InstalledPackage` and serialize to file when we do materialize this as a file...

Reply:
I can do this that way, I think there might be just some inconvenience with doing that with current go-apk API available.

I think options are: 
1. Create APK object and call [AddInstalledPackage](https://github.com/chainguard-dev/go-apk/blob/5e10ec23d8297369bc32ab1b3384cfc91d224a9b/pkg/apk/installed.go#L51) for each package. Perhaps overkill?
2. Use [PackageToInstalled](https://github.com/chainguard-dev/go-apk/blob/5e10ec23d8297369bc32ab1b3384cfc91d224a9b/pkg/apk/package.go#L31) to construct the contents of the file.

Reply:
I'm sorry. I updated my comment, as it was not clear. What I mean is to use the '[]*apk.InstalledPackage' in this object and when we need file, perform the write .

Reply:
Done (parsing installed in initialization now). Still kept the raw bytes, for convenience (no convenient []InstalledPkgs -> []byte conversion)